model/DataStore/MetaDataDeliverySyncTask.php

```diff
@@ -0,0 +1,171 @@
+use oat\oatbox\reporting\Report;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+use oat\tao\model\metadata\compiler\ResourceJsonMetadataCompiler;
+use oat\tao\model\taskQueue\QueueDispatcher;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use taoQtiTest_models_classes_QtiTestService;
+use Throwable;
+
+class MetaDataDeliverySyncTask extends AbstractAction implements JsonSerializable
+{
+
+
+    /** @var bool */
+    private $error;
+
+    /**
+     * @throws InvalidServiceManagerException
+     */
+    public function __invoke($params)
+    {
+        $params = $this->prepareData($params);
+        $report = new Report(Report::TYPE_SUCCESS);
+
+        if ($params['count'] < $params[DeliveryMetadataListener::OPTION_MAX_TRIES]) {
+            $params['count']++;
+            try {
+                $this->getPersistDataService()->persist($params);
+                $report->setMessage('Success MetaData syncing for delivery: ' . $params['deliveryId']);
+            } catch (Throwable $exception) {
+                $this->logError(sprintf(
+                    'Failing MetaData syncing for delivery: %s with message: %s',
+                    $params['deliveryId'],
+                    $exception->getMessage()
+                ));
+
+                $report->setType(Report::TYPE_ERROR);
+                $report->setMessage('Failing MetaData syncing for delivery: ' . $params['deliveryId']);
+                $this->requeueTask($params);
+            }
+        }
+
+        return $report;
+    }
+
+    public function jsonSerialize()
+    {
+        return __CLASS__;
+    }
+
+    /**
+     * @throws InvalidServiceManagerException
+     */
+    private function getQueueDispatcher(): ConfigurableService
+    {
+        return $this->getServiceManager()->get(QueueDispatcher::SERVICE_ID);
+    }
+
+    /**
+     * @param $params
+     * @throws InvalidServiceManagerException
+     */
+    private function requeueTask($params): void
+    {
+        /** @var QueueDispatcher $queueDispatcher */
+        $queueDispatcher = $this->getQueueDispatcher();
+        $queueDispatcher->createTask(
+            $this,
+            $params,
+            __('Continue try to sync GCP of delivery "%s".', $params['deliveryId'])
+        );
+    }
+
+    private function getPersistDataService(): PersistDataService
+    {
+        return $this->getServiceLocator()->get(PersistDataService::class);
+    }
+
+    /**
+     * @throws common_exception_Error
+     * @throws core_kernel_persistence_Exception
+     */
+    private function prepareData($params)
+    {
+        $params['count'] = $params['count'] ?? 0;
+        if (!isset($params['deliveryMetaData'], $params['testMetaData'], $params['testUri'], $params['itemMetaData'])) {
+            $compiler = $this->getMetaDataCompiler();
+            //DeliveryMetaData
+            $deliveryResource = new core_kernel_classes_Resource($params['deliveryId']);
```

Comment:
We should use `$this->getResource()`

Reply:
i need to use OntologyAwareTrait for that , do you think it will more optimized ?